draft-nygren-httpbis-httpssvc.md

```diff
@@ -481,6 +481,18 @@ Because HTTPSSVC is received over an often insecure channel (DNS),
 clients MUST NOT place any more trust in this signal than if they
 had received a 302 redirect over cleartext HTTP.
 
+If the HTTPSSVC query results in a SERVFAIL error, and the connection
+between the client and the recursive resolver is cryptographically protected
+(e.g. using TLS {{!RFC7858}} or HTTPS {{!RFC8484}}), the client SHOULD
+abandon the connection attempt and display an error message.  A SERVFAIL
+error can occur if the domain is DNSSEC-signed, the recursive resolver is
+DNSSEC-validating, and an active attacker between the recursive resolver
+and the authoritative DNS server is attempting to prevent the upgrade to
+HTTPS.
+
+Similarly, if the client enforces DNSSEC validation on A/AAAA
+RRs, it SHOULD abandon the connection attempt if the HTTPSSVC RR fails
+to validate.
```

Comment:
Is this special in any way? How is this different/new behavior here? (Are there clients that always enforce DNSSEC validation on all A/AAAA RRs?) While nice, I'm not sure this needs to be included here.

Reply:
The issue is that Alt-Svc and HTTPSSVC are "optional", and connection can always continue without them.  This text is intended to clarify that while their presence is optional (NXDOMAIN isn't a problem), certain failure modes should be treated as fatal, even if the connection actually would complete successfully if the client ignored the error, because this category of errors might represent adversary trying to force a downgrade.

Admittedly, this particular text is mostly academic/optimistic, since very few clients (and AFAIK no commercially deployed clients) actually validate DNSSEC on the client.